core/src/serve_repair.rs

```diff
@@ -687,6 +698,21 @@ impl ServeRepair {
                 }
             };
 
+            let staked = epoch_staked_nodes
+                .as_ref()
+                .map(|nodes| nodes.contains_key(request.sender()))
+                .unwrap_or_default();
+            if staked {
+                stats.handle_requests_staked += 1;
+            } else {
+                stats.handle_requests_unstaked += 1;
+            }
+
+            if budget_exhausted {
+                stats.dropped_requests_outbound_bandwidth += 1;
+                continue;
+            }
```

Comment:
The issue I see here is that this is still paying for deserializing packets and looking up pubkeys in `epoch_staked_nodes` hashmap even though the outbound budget is exhausted. This seems to me pretty wasteful just for the sake of collecting metrics.

The other thing, as previously mentioned: https://github.com/solana-labs/solana/pull/27771#issuecomment-1263839630
I still don't see why stake vs unstaked breakdown of load matters?
If we cannot handle staked nodes requests, then why should we allocate any budget to unstaked nodes?

Reply:
and, if we want to prioritize requests by stake, then again, we don't care about stake vs unstaked breakdown